File: problems/Single Element in a Sorted Array/singleNonDuplicate.cs
public class Solution {
    public int SingleNonDuplicate(int[] nums) {
        if (null == nums || 0 == nums.Length || 0 == (nums.Length & 1)) {
            return -1;
        }

        var n = nums.Length;
        var l = 0;
        var r = n - 1;

        while (l < r) {
            var mid = l + ((r - l) >> 1);

            if (1 == (mid & 1)) {
                --mid;
            }

            if (nums[mid] == nums[1 + mid]) {
                l = mid + 2;
            } else {
                r = mid;
            }
        }

        return nums[l];
    }
}

File: problems/Single Element in a Sorted Array/singleNonDuplicate.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int left = 0;
        int right = nums.length;

        while (left + 1 < right) {
            int mid = left + (right - left) / 2;

            if (nums[mid] != nums[mid - 1] && (1 + mid == nums.length || nums[mid] != nums[mid + 1])) {
                return nums[mid];
            } else if ((1 == (mid & 1) && nums[mid] == nums[mid - 1]) || (0 == (mid & 1) && nums[mid] == nums[mid + 1])) {
                left = mid;
            } else {
                right = mid;
            }
        }

        return nums[left];
    }
}
